packages/ui-toolkit/src/components/molecules/Calendar/MonthCalendar/index.tsx

```diff
@@ -7,49 +7,99 @@ import './monthCalendar.css';
 const MONTHS = [ 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec' ];
 
 class MonthCalendar extends React.PureComponent<Props, State> {
+  // Default props for the component
+  static defaultProps = {
+    currentDate: new Date(),
+    minMonth: null,
+    maxMonth: null,
+    onDateChange: (date: Date) => {}
+  };
+
+
+  constructor(props: Props) {
+    super(props);
+    const { currentDate, maxMonth } = props;
+
+    // Initialize state with dateToShow, considering maxMonth if provided
+    if (maxMonth) {
+      this.state = {
+        dateToShow: currentDate > maxMonth ? maxMonth : currentDate
+      };
+    }
+  }
+
+  // Initial state
 
-  state:State = {
+  state: State = {
     dateToShow: this.props.currentDate
   };
 
+  // Update state if currentDate prop changes
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.currentDate.getTime() !== this.props.currentDate.getTime()) {
+      this.setState({
+        dateToShow: this.props.currentDate
+      });
+    }
+  }
+
 
   render() {
     const { dateToShow } = this.state;
     const { currentDate } = this.props;
 
+    const isPreviousYearDisabled = this.isPreviousYearDisabled();
+    const isNextYearDisabled = this.isNextYearDisabled();
+
+    // Get the index of the current month
     const currentMonthIndex = currentDate.getMonth();
-    const presentDate = new Date();
 
     return (
-      <div className='contentPrimary'>
-        <div className='card borderPrimary mn12Box'>
+      <div className="contentPrimary">
+        <div className="card borderPrimary mn12Box">
           <div className="valign-wrapper mn12YearRow">
             <div
-              className="valign-wrapper cur-po"
-              onClick={this.handlePrevYearClick}
+              className={
+                cn({ 'valign-wrapper cur-po': !isPreviousYearDisabled }, {
+                  'contentSecondary cur-no': isPreviousYearDisabled
+                })
+              }
+              onClick={() => this.handlePrevYearClick(isPreviousYearDisabled)}
             >
-              <KeyboardArrowLeft className="mn12YearIcon"/>
+              <KeyboardArrowLeft className="mn12YearIcon" />
             </div>
             <div>{dateToShow.getFullYear()}</div>
             <div
-              className="valign-wrapper  cur-po"
-              onClick={this.handleForwardYearClick}
+              className={
+                cn({ 'valign-wrapper cur-po': !isNextYearDisabled }, {
```

Comment:
`valign-wrapper` would be a default class. Pls remove it from conditional rendering.